sdks/python/apache_beam/io/kinesis.py

```diff
@@ -0,0 +1,317 @@
+
+  There are several ways to setup cross-language Kinesis transforms.
+
+  * Option 1: use the default expansion service
+  * Option 2: specify a custom expansion service
+
+  See below for details regarding each of these options.
+
+  *Option 1: Use the default expansion service*
+
+  This is the recommended and easiest setup option for using Python Kinesis
+  transforms. This option is only available for Beam 2.24.0 and later.
+
+  This option requires following pre-requisites before running the Beam
+  pipeline.
+
+  * Install Java runtime in the computer from where the pipeline is constructed
+    and make sure that 'java' command is available.
+
+  In this option, Python SDK will either download (for released Beam version) or
+  build (when running from a Beam Git clone) a expansion service jar and use
+  that to expand transforms. Currently Kinesis transforms use the
+  'beam-sdks-java-io-kinesis-expansion-service' jar for this purpose.
+
+  *Option 2: specify a custom expansion service*
+
+  In this option, you startup your own expansion service and provide that as
+  a parameter when using the transforms provided in this module.
+
+  This option requires following pre-requisites before running the Beam
+  pipeline.
+
+  * Startup your own expansion service.
+  * Update your pipeline to provide the expansion service address when
+    initiating Kinesis transforms provided in this module.
+
+  Flink Users can use the built-in Expansion Service of the Flink Runner's
+  Job Server. If you start Flink's Job Server, the expansion service will be
+  started on port 8097. For a different address, please set the
+  expansion_service parameter.
+
+  **More information**
+
+  For more information regarding cross-language transforms see:
+  - https://beam.apache.org/roadmap/portability/
+
+  For more information specific to Flink runner see:
+  - https://beam.apache.org/documentation/runners/flink/
+"""
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import time
+from typing import List
+from typing import NamedTuple
+from typing import Optional
+from typing import Tuple
+
+from past.builtins import unicode
+
+from apache_beam import BeamJarExpansionService
+from apache_beam import ExternalTransform
+from apache_beam import NamedTupleBasedPayloadBuilder
+
+__all__ = [
+    'WriteToKinesis',
+    'ReadDataFromKinesis',
+    'InitialPositionInStream',
+    'WatermarkPolicy',
+]
+
+
+def default_io_expansion_service():
+  return BeamJarExpansionService(
+      'sdks:java:io:kinesis:expansion-service:shadowJar')
+
+
+WriteToKinesisSchema = NamedTuple(
+    'WriteToKinesisSchema',
+    [
+        ('stream_name', unicode),
+        ('aws_access_key', unicode),
+        ('aws_secret_key', unicode),
+        ('region', unicode),
+        ('partition_key', unicode),
+        ('service_endpoint', Optional[unicode]),
+        ('verify_certificate', Optional[bool]),
+        ('producer_properties', Optional[List[Tuple[unicode, unicode]]]),
```

Comment:
FYI if https://github.com/apache/beam/pull/12481 is merged first this can (and will have to) change to `Mapping[unicode, unicode]`

Reply:
Great, Mapping will at least be there! I'll follow this PR.

Reply:
Done. I also added some config params in the test to verify that Mapping works.